dali/kernels/imgproc/color_manipulation/brightness_contrast_gpu.h

```diff
@@ -0,0 +1,211 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_KERNELS_IMGPROC_COLOR_MANIPULATION_BRIGHTNESS_CONTRAST_GPU_H_
+#define DALI_KERNELS_IMGPROC_COLOR_MANIPULATION_BRIGHTNESS_CONTRAST_GPU_H_
+
+#include <vector>
+#include "dali/core/convert.h"
+#include "dali/core/geom/box.h"
+#include "dali/kernels/common/block_setup.h"
+
+namespace dali {
+namespace kernels {
+namespace brightness_contrast {
+
+
+template <size_t ndims>
+using Roi = Box<ndims, int>;
+
+
+template <class InputType, class OutputType, int ndims>
+struct SampleDescriptor {
+  const InputType *in;
+  OutputType *out;
+  ivec<ndims> in_pitch, out_pitch;
+  float brightness, contrast;
+};
+
+
+template <size_t ndims>
+TensorListShape<ndims> RoiToShape(const std::vector<Roi<ndims>> &rois, int nchannels) {
+  std::vector<TensorShape<ndims>> ret;
+
+  for (auto roi : rois) {
+    assert(all_coords(roi.hi >= roi.lo) && "Cannot create a tensor shape from an invalid Box");
```

Comment:
I used this, because it's only debug info for me. This is checked earlier, so I don't think it's neccessary to check it again. And `assert` is harmless

Reply:
Well, it's debug info. No included in `Release` build, this shows explicitly the asserts of function, so has also documentation purpose

Reply:
yes, but since you area already checking this at a higher level, I think it just adds unnecessary clutter